src/types.ts

```diff
@@ -0,0 +1,556 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// @ts-ignore
+import hathorLib from '@hathor/wallet-lib';
+import { isAuthority } from './utils';
+
+export interface EventTxInput {
+  tx_id: string;
+  index: number;
+  spent_output: EventTxOutput;
+}
+
+export interface EventTxOutput {
+  value: number;
+  token_data: number;
+  script: string;
+  locked?: boolean;
+  decoded: {
+    type: string;
+    address: string;
+    timelock: number | null;
+  };
+}
+
+export interface DecodedOutput {
+  type: string;
+  address: string;
+  timelock: number | null;
+}
+
+export interface TxOutput {
+  value: number;
+  script: string;
+  token: string;
+  decoded: DecodedOutput | null;
+  // eslint-disable-next-line camelcase
+  spent_by?: string | null;
+  // eslint-disable-next-line camelcase
+  token_data: number;
+  locked?: boolean;
+}
+
+export interface DbTxOutput {
+  txId: string;
+  index: number;
+  tokenId: string;
+  address: string;
+  value: number;
+  authorities: number;
+  timelock: number | null;
+  heightlock: number | null;
+  locked: boolean;
+  spentBy?: string | null;
+  txProposalId?: string | null;
+  txProposalIndex?: number | null;
+  voided?: boolean | null;
+}
+
+export interface TxOutputWithIndex extends TxOutput {
+  index: number;
+}
+
+export interface TxInput {
+  // eslint-disable-next-line camelcase
+  tx_id: string;
+  index: number;
+  value: number;
+  // eslint-disable-next-line camelcase
+  token_data: number;
+  script: string;
+  token: string;
+  decoded: DecodedOutput | null;
+}
+
+export interface StringMap<T> {
+  [x: string]: T;
+}
```

Comment:
Not really requesting a change, but is there any difference between this and:
```ts
export interface StringMap<T> = Record<string, T>;
```

I'm just wondering because map and plain object have some differences that can cause some confusion.